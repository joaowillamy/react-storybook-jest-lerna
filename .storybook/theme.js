import { create } from "@storybook/theming/create";

export default create({
  base: "light",

  colorPrimary: "#FDCB13",
  colorSecondary: "#18ACC4",

  // UI
  appBg: "white",
  appContentBg: "#F2F3F4",
  appBorderColor: "#F2F3F4",
  appBorderRadius: 4,

  // Typography
  fontBase: '"Montserrat", sans-serif',
  fontCode: "monospace",

  brandTitle: "Quero front test"
  // brandUrl: 'http://resolvvi.com.br/',
  // brandImage: ''
});
