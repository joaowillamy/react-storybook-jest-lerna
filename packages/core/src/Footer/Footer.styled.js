import styled, { css } from "styled-components";
import media from "styled-media-query";

import { chooseScrem } from "../help/chooseScreem";

export const Footer = styled.div`
  background-color: ${({ theme }) => theme.colors.blueSecondary};
`;

export const FooterEnd = styled.div`
  background-color: ${({ theme }) => theme.colors.bluePrimary};
  padding: ${({ theme }) => theme.space.xxxl} 0;
  color: #fff;
  font-weight: 600;
  text-align: center;
  font-size: 12px;
`;

export const FooterCards = styled.ul`
  width: 100%;
`;

export const FooterCardContainer = styled.div`
  align-items: center;
  display: flex;
  justify-content: center;

  ${media.lessThan("medium")`
    ${({ fullInMobile }) => !fullInMobile && "display: block;"}
    `}
`;

export const FooterCardText = styled.span`
  font-weight: 600;
  font-size: 13px;
  ${chooseScrem}
`;

export const FooterCardSubText = styled.span`
  font-weight: 300;
  font-size: 12px;
  ${chooseScrem}
`;

export const FooterCard = styled.li`
  display: inline-block;
  padding: 20px;
  color: #fff;
  font-size: 13px;
  text-align: center;

  svg {
    width: 26px;
    color: #fff;
    margin-bottom: 10px;
  }

  ${({ borderInMobile, theme }) =>
    borderInMobile &&
    css`
      border-right: 4px solid ${theme.colors.bluePrimary};
      border-left: 4px solid ${theme.colors.bluePrimary};
    `}

  ${({ fullInMobile, theme }) => css`
    width: ${fullInMobile ? "100%" : "33%"};

    ${fullInMobile &&
      css`
        border-bottom: 4px solid ${theme.colors.bluePrimary};
        text-align: left;
        svg {
          margin-right: 10px;
          margin-bottom: 0px;
        }
      `}
  `};

  ${media.greaterThan("medium")`
    border:none;
    width: 25%;
    text-align: left;
    svg {
          margin-right: 10px;
          margin-bottom: 0px;
        }
  `}
`;
