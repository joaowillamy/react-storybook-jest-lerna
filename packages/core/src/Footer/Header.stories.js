import React from 'react';

import Footer from '.';

export default {
  component: Footer,
  title: 'Core|Footer'
};

export const primary = () => <Footer>{'Footer'}</Footer>;

export const secondary = () => <Footer variant="secondary">{'Footer'}</Footer>;
