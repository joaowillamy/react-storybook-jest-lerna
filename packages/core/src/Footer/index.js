import React from "react";

import { Icon } from "../Icons";
import { Container } from "../help/Container.styled";

import * as S from "./Footer.styled";

const Footer = () => (
  <S.Footer>
    <footer>
      <Container>
        <S.FooterCards>
          <S.FooterCard fullInMobile>
            <S.FooterCardContainer fullInMobile>
              <Icon.Whatsapp />
              <div>
                <S.FooterCardText>{"0800 123 2222"}</S.FooterCardText> <br />
                <S.FooterCardSubText justMobile>
                  {"Segunda a sexta de 8h às 22h"}
                </S.FooterCardSubText>
                <S.FooterCardSubText justDesktop>
                  {"Seg - Sex 8h-22h"}
                </S.FooterCardSubText>
              </div>
            </S.FooterCardContainer>
          </S.FooterCard>

          <S.FooterCard>
            <S.FooterCardContainer>
              <Icon.Chatbubbles />
              <div>
                <S.FooterCardText justMobile>{"Chat"}</S.FooterCardText>
                <S.FooterCardText justDesktop>
                  {"Chat ao vivo"}
                </S.FooterCardText>{" "}
                <br />
                <S.FooterCardSubText justDesktop>
                  {"Seg - Sex 8h-22h"}
                </S.FooterCardSubText>
              </div>
            </S.FooterCardContainer>
          </S.FooterCard>

          <S.FooterCard borderInMobile>
            <S.FooterCardContainer>
              <Icon.Email />
              <div>
                <S.FooterCardText justMobile>{"E-mail"}</S.FooterCardText>
                <S.FooterCardText justDesktop>
                  {"Mande um e-mail"}
                </S.FooterCardText>{" "}
                <br />
                <S.FooterCardSubText justDesktop>
                  {"Respondemos rapidinho"}
                </S.FooterCardSubText>
              </div>
            </S.FooterCardContainer>
          </S.FooterCard>

          <S.FooterCard>
            <S.FooterCardContainer>
              <Icon.InfoCircle />
              <div>
                <S.FooterCardText justMobile>{"Ajuda"}</S.FooterCardText>
                <S.FooterCardText justDesktop>
                  {"Central de ajuda"}
                </S.FooterCardText>{" "}
                <br />
                <S.FooterCardSubText justDesktop>
                  {"Encontre todas as respostas"}
                </S.FooterCardSubText>
              </div>
            </S.FooterCardContainer>
          </S.FooterCard>
        </S.FooterCards>
      </Container>
      <S.FooterEnd>{"Feito com ❤️ pela Quero Educação"}</S.FooterEnd>
    </footer>
  </S.Footer>
);

export default Footer;
