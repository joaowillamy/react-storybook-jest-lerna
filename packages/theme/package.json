{
  "name": "@will-ui-kit/theme",
  "version": "1.0.2",
  "main": "lib/index.js",
  "module": "src/index.js"
}
