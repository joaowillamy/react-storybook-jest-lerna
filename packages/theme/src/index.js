export { default as ThemeProvider } from "./ThemeProvider";
export { default as DEFAULT_THEME } from "./theme";
export { GlobalStyles } from "./GlobalStyles";
