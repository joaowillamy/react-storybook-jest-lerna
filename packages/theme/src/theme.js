const BASE = 3;

const borders = {
  sm: color => `1px solid ${color || "#00000029"}`,
  md: color => `3px solid ${color || "#00000029"}`
};

const colors = {
  bluePrimary: "#18ACC4",
  blueSecondary: "#007A8D",
  yellowPrimary: "#FDCB13",
  yellowSecondary: "#DE9E1F",
  green: "#0FA866",
  black: "#1F2D30",
  grey: "#FBFBFB",
  Overlay: "#1F2D30e0" // com 88% de opacidade.
};

const fonts = {
  Montserrat: "Montserrat"
};

const fontSizes = {
  xs: "12px",
  sm: "14px",
  md: "16px",
  lg: "18px",
  xl: "22px",
  xxl: "26px",
  xxxl: "36px"
};

const fontWeights = {
  thin: 100,
  extralight: 200,
  light: 300,
  regular: 400,
  medium: 500,
  semibold: 600,
  bold: 700,
  extrabold: 800,
  black: 900
};

const iconSizes = {
  sm: "12px",
  md: "16px",
  lg: "26px"
};

const shadows = {
  sm: color => `0 1px 3px ${color || "#00000029"}`,
  md: color => `0 3px 4px ${color || "#00000029"}`,
  lg: color => `0 3px 6px ${color || "#00000029"}`
};

const space = {
  xs: `${BASE}px`,
  sm: `${BASE * 2}px`,
  md: `${BASE * 3}px`,
  lg: `${BASE * 4}px`,
  xl: `${BASE * 8}px`,
  xxl: `${BASE * 10}px`,
  xxxl: `${BASE * 12}px`
};

const DEFAULT_THEME = {
  borders,
  colors,
  fonts,
  fontSizes,
  fontWeights,
  iconSizes,
  shadows,
  space
};

export default DEFAULT_THEME;
