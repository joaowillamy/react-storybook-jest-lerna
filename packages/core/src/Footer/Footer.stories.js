import React from "react";
import { ThemeProvider } from "@joaowillamy-test-quero/theme";

import Footer from ".";

export default {
  component: Footer,
  title: "Core|Footer"
};

export const footer = () => (
  <ThemeProvider>
    <Footer />
  </ThemeProvider>
);
