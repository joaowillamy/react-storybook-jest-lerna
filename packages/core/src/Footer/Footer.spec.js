import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { ThemeProvider } from "@joaowillamy-test-quero/theme";

import Footer from ".";

describe("Footer", () => {
  test("Footer", () => {
    const tree = renderer
      .create(
        <ThemeProvider>
          <Footer />
        </ThemeProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
