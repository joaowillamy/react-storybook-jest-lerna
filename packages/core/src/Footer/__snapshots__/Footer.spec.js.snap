// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Footer 1`] = `
.c5 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  background-color: #007A8D;
}

.c14 {
  background-color: #18ACC4;
  padding: 36px 0;
  color: #fff;
  font-weight: 600;
  text-align: center;
  font-size: 12px;
}

.c2 {
  width: 100%;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c10 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  font-weight: 600;
  font-size: 13px;
}

.c11 {
  font-weight: 600;
  font-size: 13px;
}

.c12 {
  font-weight: 600;
  font-size: 13px;
}

.c7 {
  font-weight: 300;
  font-size: 12px;
}

.c8 {
  font-weight: 300;
  font-size: 12px;
}

.c3 {
  display: inline-block;
  padding: 20px;
  color: #fff;
  font-size: 13px;
  text-align: center;
  width: 100%;
  border-bottom: 4px solid #18ACC4;
  text-align: left;
}

.c3 svg {
  width: 26px;
  color: #fff;
  margin-bottom: 10px;
}

.c3 svg {
  margin-right: 10px;
  margin-bottom: 0px;
}

.c9 {
  display: inline-block;
  padding: 20px;
  color: #fff;
  font-size: 13px;
  text-align: center;
  width: 33%;
}

.c9 svg {
  width: 26px;
  color: #fff;
  margin-bottom: 10px;
}

.c13 {
  display: inline-block;
  padding: 20px;
  color: #fff;
  font-size: 13px;
  text-align: center;
  border-right: 4px solid #18ACC4;
  border-left: 4px solid #18ACC4;
  width: 33%;
}

.c13 svg {
  width: 26px;
  color: #fff;
  margin-bottom: 10px;
}

@media (min-width:768px) {
  .c1 {
    width: 95%;
    margin: 0 auto;
  }
}

@media (max-width:768px) {

}

@media (max-width:768px) {
  .c10 {
    display: block;
  }
}

@media (min-width:768px) {
  .c11 {
    display: none;
  }
}

@media (max-width:768px) {
  .c12 {
    display: none;
  }
}

@media (min-width:768px) {
  .c7 {
    display: none;
  }
}

@media (max-width:768px) {
  .c8 {
    display: none;
  }
}

@media (min-width:768px) {
  .c3 {
    border: none;
    width: 25%;
    text-align: left;
  }

  .c3 svg {
    margin-right: 10px;
    margin-bottom: 0px;
  }
}

@media (min-width:768px) {
  .c9 {
    border: none;
    width: 25%;
    text-align: left;
  }

  .c9 svg {
    margin-right: 10px;
    margin-bottom: 0px;
  }
}

@media (min-width:768px) {
  .c13 {
    border: none;
    width: 25%;
    text-align: left;
  }

  .c13 svg {
    margin-right: 10px;
    margin-bottom: 0px;
  }
}

<div
  className="c0"
>
  <footer>
    <div
      className="c1"
    >
      <ul
        className="c2"
      >
        <li
          className="c3"
        >
          <div
            className="c4"
          >
            <svg
              aria-hidden="true"
              className="c5"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M380.9 97.1C339 55.1 283.2 32 223.9 32c-122.4 0-222 99.6-222 222 0 39.1 10.2 77.3 29.6 111L0 480l117.7-30.9c32.4 17.7 68.9 27 106.1 27h.1c122.3 0 224.1-99.6 224.1-222 0-59.3-25.2-115-67.1-157zm-157 341.6c-33.2 0-65.7-8.9-94-25.7l-6.7-4-69.8 18.3L72 359.2l-4.4-7c-18.5-29.4-28.2-63.3-28.2-98.2 0-101.7 82.8-184.5 184.6-184.5 49.3 0 95.6 19.2 130.4 54.1 34.8 34.9 56.2 81.2 56.1 130.5 0 101.8-84.9 184.6-186.6 184.6zm101.2-138.2c-5.5-2.8-32.8-16.2-37.9-18-5.1-1.9-8.8-2.8-12.5 2.8-3.7 5.6-14.3 18-17.6 21.8-3.2 3.7-6.5 4.2-12 1.4-32.6-16.3-54-29.1-75.5-66-5.7-9.8 5.7-9.1 16.3-30.3 1.8-3.7.9-6.9-.5-9.7-1.4-2.8-12.5-30.1-17.1-41.2-4.5-10.8-9.1-9.3-12.5-9.5-3.2-.2-6.9-.2-10.6-.2-3.7 0-9.7 1.4-14.8 6.9-5.1 5.6-19.4 19-19.4 46.3 0 27.3 19.9 53.7 22.6 57.4 2.8 3.7 39.1 59.7 94.8 83.8 35.2 15.2 49 16.5 66.6 13.9 10.7-1.6 32.8-13.4 37.4-26.4 4.6-13 4.6-24.1 3.2-26.4-1.3-2.5-5-3.9-10.5-6.6z"
                fill="currentColor"
              />
            </svg>
            <div>
              <span
                className="c6"
              >
                0800 123 2222
              </span>
               
              <br />
              <span
                className="c7"
              >
                Segunda a sexta de 8h às 22h
              </span>
              <span
                className="c8"
              >
                Seg - Sex 8h-22h
              </span>
            </div>
          </div>
        </li>
        <li
          className="c9"
        >
          <div
            className="c10"
          >
            <svg
              aria-hidden="true"
              className="c5"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M431 320.6c-1-3.6 1.2-8.6 3.3-12.2a33.68 33.68 0 012.1-3.1A162 162 0 00464 215c.3-92.2-77.5-167-173.7-167-83.9 0-153.9 57.1-170.3 132.9a160.7 160.7 0 00-3.7 34.2c0 92.3 74.8 169.1 171 169.1 15.3 0 35.9-4.6 47.2-7.7s22.5-7.2 25.4-8.3a26.44 26.44 0 019.3-1.7 26 26 0 0110.1 2l56.7 20.1a13.52 13.52 0 003.9 1 8 8 0 008-8 12.85 12.85 0 00-.5-2.7z"
                fill="none"
                stroke="currentColor"
                strokeLinecap="round"
                strokeMiterlimit={10}
                strokeWidth={32}
              />
              <path
                d="M66.46 232a146.23 146.23 0 006.39 152.67c2.31 3.49 3.61 6.19 3.21 8s-11.93 61.87-11.93 61.87a8 8 0 002.71 7.68A8.17 8.17 0 0072 464a7.26 7.26 0 002.91-.6l56.21-22a15.7 15.7 0 0112 .2c18.94 7.38 39.88 12 60.83 12A159.21 159.21 0 00284 432.11"
                fill="none"
                stroke="currentColor"
                strokeLinecap="round"
                strokeMiterlimit={10}
                strokeWidth={32}
              />
            </svg>
            <div>
              <span
                className="c11"
              >
                Chat
              </span>
              <span
                className="c12"
              >
                Chat ao vivo
              </span>
               
              <br />
              <span
                className="c8"
              >
                Seg - Sex 8h-22h
              </span>
            </div>
          </div>
        </li>
        <li
          className="c13"
        >
          <div
            className="c10"
          >
            <svg
              aria-hidden="true"
              className="c5"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6zm-2 0l-8 5-8-5h16zm0 12H4V8l8 5 8-5v10z"
              />
            </svg>
            <div>
              <span
                className="c11"
              >
                E-mail
              </span>
              <span
                className="c12"
              >
                Mande um e-mail
              </span>
               
              <br />
              <span
                className="c8"
              >
                Respondemos rapidinho
              </span>
            </div>
          </div>
        </li>
        <li
          className="c9"
        >
          <div
            className="c10"
          >
            <svg
              aria-hidden="true"
              className="c5"
              fill="currentColor"
              focusable="false"
              viewBox="0 0 16 16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 15A7 7 0 108 1a7 7 0 000 14zm0 1A8 8 0 108 0a8 8 0 000 16z"
                fillRule="evenodd"
              />
              <path
                d="M8.93 6.588l-2.29.287-.082.38.45.083c.294.07.352.176.288.469l-.738 3.468c-.194.897.105 1.319.808 1.319.545 0 1.178-.252 1.465-.598l.088-.416c-.2.176-.492.246-.686.246-.275 0-.375-.193-.304-.533L8.93 6.588z"
              />
              <circle
                cx={8}
                cy={4.5}
                r={1}
              />
            </svg>
            <div>
              <span
                className="c11"
              >
                Ajuda
              </span>
              <span
                className="c12"
              >
                Central de ajuda
              </span>
               
              <br />
              <span
                className="c8"
              >
                Encontre todas as respostas
              </span>
            </div>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="c14"
    >
      Feito com ❤️ pela Quero Educação
    </div>
  </footer>
</div>
`;
